### PR TITLE
CMM 9 cards

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/for_the_ancestors.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Name:For the Ancestors
+ManaCost:2 G
+Types:Instant
+A:SP$ ChooseType | Defined$ You | Type$ Creature | SubAbility$ DBDig | AILogic$ MostProminentInComputerDeckNonToken | SpellDescription$ Choose a creature type. Look at the top six cards of your library. You may reveal any number of cards of the chosen type from among them and put the revealed cards into your hand. Put the rest on the bottom of your library in a random order.
+SVar:DBDig:DB$ Dig | DigNum$ 6 | Reveal$ True | AnyNumber$ True | ChangeValid$ Artifact | ChangeNum$ Any | ChangeValid$ Creature.ChosenType | ForceRevealToController$ True | DestinationZone$ Hand | RestRandomOrder$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/for_the_ancestors.txt:5]: ChangeValid twice

File: forge-gui/res/cardsfolder/upcoming/rukarumal_biologist.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Name:Rukarumel, Biologist
+ManaCost:W U B R G
+Types:Legendary Creature Human Wizard
+PT:3/3
+K:ETBReplacement:Other:ChooseCT
+SVar:ChooseCT:DB$ ChooseType | Defined$ You | Type$ Creature | SpellDescription$ As CARDNAME enters the battlefield, choose a creature type. | AILogic$ MostProminentInComputerDeck
+S:Mode$ Continuous | Affected$ Creature.YouCtrl+nonToken,Sliver.YouCtrl | AffectedZone$ Battlefield,Hand,Graveyard,Exile,Stack,Library,Command | AddType$ ChosenType | Description$ Slivers you control and nontoken creatures you control are the chosen type in addition to their other creature types. The same is true for creature spells you control and creature cards you own that aren’t on the battlefield.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/rukarumal_biologist.txt:7]: curly

File: forge-gui/res/cardsfolder/upcoming/titan_of_littjara.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+Name:Titan of Littjara
+ManaCost:4 U U
+Types:Creature Illusion
+PT:6/6
+K:ETBReplacement:Other:ChooseCT
+SVar:ChooseCT:DB$ ChooseType | Type$ Creature | AILogic$ MostProminentInComputerDeck | SpellDescription$ As CARDNAME enters the battlefield, choose a creature type.
+S:Mode$ Continuous | EffectZone$ All | Affected$ Card.Self | CharacteristicDefining$ True | AddType$ ChosenType | Description$ CARDNAME is the chosen type in addition to its other types.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/titan_of_littjara.txt:7]: @Hanmac 
is this really CDA?

File: forge-gui/res/cardsfolder/upcoming/titan_of_littjara.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+Name:Titan of Littjara
+ManaCost:4 U U
+Types:Creature Illusion
+PT:6/6
+K:ETBReplacement:Other:ChooseCT
+SVar:ChooseCT:DB$ ChooseType | Type$ Creature | AILogic$ MostProminentInComputerDeck | SpellDescription$ As CARDNAME enters the battlefield, choose a creature type.
+S:Mode$ Continuous | EffectZone$ All | Affected$ Card.Self | CharacteristicDefining$ True | AddType$ ChosenType | Description$ CARDNAME is the chosen type in addition to its other types.
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigDraw | TriggerDescription$ Whenever CARDNAME enters the battlefield or attacks, you may draw a card for each other creature you control that shares a creature type with it. If you do, discard a card.
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigDraw | Secondary$ True | TriggerDescription$ Whenever CARDNAME enters the battlefield or attacks, you may draw a card for each other creature you control that shares a creature type with it. If you do, discard a card.
+SVar:TrigDraw:DB$ Draw | NumCards$ X | Optional$ True | SubAbility$ DBDiscard
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/titan_of_littjara.txt:10]: maybe write as Cost
the X is also wrong

File: forge-gui/res/cardsfolder/upcoming/titan_of_littjara.txt [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+ManaCost:4 U U
+Types:Creature Illusion
+PT:6/6
+K:ETBReplacement:Other:ChooseCT
+SVar:ChooseCT:DB$ ChooseType | Type$ Creature | AILogic$ MostProminentInComputerDeck | SpellDescription$ As CARDNAME enters the battlefield, choose a creature type.
+S:Mode$ Continuous | EffectZone$ All | Affected$ Card.Self | CharacteristicDefining$ True | AddType$ ChosenType | Description$ CARDNAME is the chosen type in addition to its other types.
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigDraw | TriggerDescription$ Whenever CARDNAME enters the battlefield or attacks, you may draw a card for each other creature you control that shares a creature type with it. If you do, discard a card.
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigDraw | Secondary$ True | TriggerDescription$ Whenever CARDNAME enters the battlefield or attacks, you may draw a card for each other creature you control that shares a creature type with it. If you do, discard a card.
+SVar:TrigDraw:AB$ Discard | Cost$ Draw<X/You> | Defined$ You | NumCards$ 1 | Mode$ TgtChoose
+SVar:X:Count$Valid Creature.YouCtrl+Other+SharesCardTypeWith Card.Self
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/titan_of_littjara.txt:11]: it says creature type not card type...

File: forge-gui/res/cardsfolder/upcoming/for_the_ancestors.txt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+Name:For the Ancestors
+ManaCost:2 G
+Types:Instant
+A:SP$ ChooseType | Defined$ You | Type$ Creature | SubAbility$ DBDig | AILogic$ MostProminentInComputerDeckNonToken | SpellDescription$ Choose a creature type. Look at the top six cards of your library. You may reveal any number of cards of the chosen type from among them and put the revealed cards into your hand. Put the rest on the bottom of your library in a random order.
+SVar:DBDig:DB$ Dig | DigNum$ 6 | Reveal$ True | AnyNumber$ True | ChangeNum$ Any | ChangeValid$ Creature.ChosenType | ForceRevealToController$ True | DestinationZone$ Hand | RestRandomOrder$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/for_the_ancestors.txt:5]: should only reveal cards of the type

File: forge-gui/res/cardsfolder/upcoming/titan_of_littjara.txt [hunk 7]
```diff
@@ -0,0 +1,13 @@
+Name:Titan of Littjara
+ManaCost:4 U U
+Types:Creature Illusion
+PT:6/6
+K:ETBReplacement:Other:ChooseCT
+SVar:ChooseCT:DB$ ChooseType | Type$ Creature | AILogic$ MostProminentInComputerDeck | SpellDescription$ As CARDNAME enters the battlefield, choose a creature type.
+S:Mode$ Continuous | EffectZone$ All | Affected$ Card.Self | AddType$ ChosenType | Description$ CARDNAME is the chosen type in addition to its other types.
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/upcoming/titan_of_littjara.txt:7]: Not EffectZone All

File: forge-gui/res/cardsfolder/upcoming/titan_of_littjara.txt [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+ManaCost:4 U U
+Types:Creature Illusion
+PT:6/6
+K:ETBReplacement:Other:ChooseCT
+SVar:ChooseCT:DB$ ChooseType | Type$ Creature | AILogic$ MostProminentInComputerDeck | SpellDescription$ As CARDNAME enters the battlefield, choose a creature type.
+S:Mode$ Continuous | EffectZone$ All | Affected$ Card.Self | AddType$ ChosenType | Description$ CARDNAME is the chosen type in addition to its other types.
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigDraw | TriggerDescription$ Whenever CARDNAME enters the battlefield or attacks, you may draw a card for each other creature you control that shares a creature type with it. If you do, discard a card.
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigDraw | Secondary$ True | TriggerDescription$ Whenever CARDNAME enters the battlefield or attacks, you may draw a card for each other creature you control that shares a creature type with it. If you do, discard a card.
+SVar:TrigDraw:AB$ Discard | Cost$ Draw<X/You> | Defined$ You | NumCards$ 1 | Mode$ TgtChoose
+SVar:X:Count$Valid Creature.YouCtrl+Other+sharesCreatureTypeWith Self
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/titan_of_littjara.txt:11]: Self is redundant

File: forge-gui/res/cardsfolder/upcoming/for_the_ancestors.txt [hunk 9]
```diff
@@ -0,0 +1,8 @@
+Name:For the Ancestors
+ManaCost:2 G
+Types:Instant
+A:SP$ ChooseType | Defined$ You | Type$ Creature | SubAbility$ DBDig | AILogic$ MostProminentInComputerDeckNonToken | SpellDescription$ Choose a creature type. Look at the top six cards of your library. You may reveal any number of cards of the chosen type from among them and put the revealed cards into your hand. Put the rest on the bottom of your library in a random order.
+SVar:DBDig:DB$ Dig | DigNum$ 6 | AnyNumber$ True | ChangeNum$ Any | ChangeValid$ Creature.ChosenType | DestinationZone$ Hand | RestRandomOrder$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/for_the_ancestors.txt:5]: ChangeNum$ Any does not exist

**Review by tool4ever — APPROVED** [review]

**Pull request merged**